### PR TITLE
Queueable job from configuration file

### PR DESCRIPTION
**Pull request by masterix21**
**NOTE:** the issue is strictly linked to https://github.com/spatie/laravel-multitenancy/pull/78

⚠️ **WARNING** ⚠️ the PR breaks the installation without the config parameter `multitenancy.queueable_to_job`.

Using the package laravel-actions by Loris Leiva, I've found a limit to resolving his JobDecoration.

Before changing the code, I've created the tests for the code introduced in the #78 PR: it has been done to reduce the errors in any future changes.

Finally, I've moved the switch cases to the config file. Now the package can resolve queuable properties or methods, giving the liberty to support any job.

### CODE REVIEW

**Comment by masterix21** [comment]
@freekmurze, I think we can merge it but, before, I'd like to hear your opinion.

Thanks

File: src/Actions/MakeQueueTenantAwareAction.php [hunk 1]
```diff
@@ -88,17 +89,14 @@ protected function findTenant(JobProcessing $event): Tenant
...
-                return $queueable->class;
-            case BroadcastEvent::class:
-                return $queueable->event;
-            default:
-                return $queueable;
+        $job = Arr::get(config('multitenancy.queueable_to_job'), get_class($queueable));
+
+        if (! $job) {
+            return $queueable;
+        } elseif (method_exists($queueable, $job)) {
```
Comment (freekmurze) [inline src/Actions/MakeQueueTenantAwareAction.php:96]: Do we need `else` here? Couldn't it be a regular `if`?

**Comment by freekmurze** [comment]
Just one tiny comment, seems good to merge otherwise!

File: src/Actions/MakeQueueTenantAwareAction.php (hunk 1 above)
Comment (masterix21) [reply src/Actions/MakeQueueTenantAwareAction.php:96]: Ok ;)

**Comment by masterix21** [comment]
Great ;)

**Pull request merged**